### PR TITLE
Create users repository and inject it in Program.cs

### PR DESCRIPTION
**Pull request by IamBeaver**
Create an interface for users repository, implement it and inject him in Program.cs. Add automapper and two dtos for user entity.

### CODE REVIEW

File: Backend/Backend/Repository/UsersRepository.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            return true;
+        }
+
+        public async Task<IEnumerable<User>> Read()
+        {
+            return await _dbContext.Users.ToListAsync();
+        }
+
+        public async Task<bool> Update(int id, string firstName, string lastName, int age)
+        {
```
Comment (anniemt6) [inline Backend/Backend/Repository/UsersRepository.cs:51]: Too many passed parameters. It's better to pass less than 3 inclusive. Maybe you should pass here argument with type User.

File: Backend/Backend/Repository/UsersRepository.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            await _dbContext.SaveChangesAsync();
+            return user.Entity.Id;
+        }
+
+        public async Task<bool> Delete(int id)
+        {
+            try
+            {
+                _dbContext.Users.Remove(await _dbContext.Users.FirstOrDefaultAsync(x => x.Id == id) ?? throw new NullReferenceException("User wasn't found"));
+                await _dbContext.SaveChangesAsync();
```
Comment (anniemt6) [inline Backend/Backend/Repository/UsersRepository.cs:35]: Only one error handled. What about the rest of the errors associated with the methods? Maybe it's better to handle errors at another level of abstraction? Or you can process them right away, but handle all.

File: Backend/Backend/Repository/UsersRepository.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            }
+            catch (NullReferenceException)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public async Task<IEnumerable<User>> Read()
```
Comment (MarikNez) [inline Backend/Backend/Repository/UsersRepository.cs]: Naming - hard to understand what method do

File: Backend/Backend/Repository/UsersRepository.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+        public async Task<bool> Update(int id, string firstName, string lastName, int age)
+        {
+            var user = await _dbContext.Users.FirstOrDefaultAsync(x => x.Id == id);
+            if (user is null)
+            {
+                return false;
+            }
+
+            user.FirstName = firstName;
```
Comment (MarikNez) [inline Backend/Backend/Repository/UsersRepository.cs]: Try to add AutoMapper and use it for mapping

File: Backend/Backend/Repository/UsersRepository.cs (hunk 2 above)
Comment (IamBeaver) [reply Backend/Backend/Repository/UsersRepository.cs:35]: I don't understand clearly what do you mean. Is it bad to handle only one error here? Which errors i have to handle else?

**Pull request merged**